### PR TITLE
feat(warthog-service): adds base warthog service

### PR DESCRIPTION
**Pull request by goldcaddy77**

### CODE REVIEW

File: src/tgql/BaseResolver.ts [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
+    // // TODO: This handles an issue with typeorm-typedi-extensions where it is unable to
+    // // Inject the proper repository
+    // if (!repository) {
+    //   this.repository = getRepository(entityClass);
+    // }
+    // if (!repository) {
+    //   throw new Error(`BaseResolver requires a valid repository, class ${entityClass}`);
+    // }
+
+    this.service = new WarthogService<E>(entityClass, this.repository);
```
Comment (goldcaddy77) [inline src/tgql/BaseResolver.ts]: Now the base resolver creates a service and proxies  all requests through.

File: examples/6-base-service/src/user.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  UserUpdateArgs,
+  UserWhereArgs,
+  UserWhereInput,
+  UserWhereUniqueInput
+} from '../generated';
+
+import { User } from './user.model';
+
+@Resolver(User)
+export class UserResolver {
```
Comment (goldcaddy77) [inline examples/6-base-service/src/user.resolver.ts]: This is an example that doesn't use BaseResolver.

File: examples/6-base-service/src/user.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  service: WarthogService<User>;
+
+  constructor(@InjectRepository(User) private readonly userRepository: Repository<User>) {
+    this.service = new WarthogService<User>(User, userRepository);
```
Comment (goldcaddy77) [inline examples/6-base-service/src/user.resolver.ts]: Instead, we spin up a service that will make all of the generic calls.

File: examples/6-base-service/src/user.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    @Inject() readonly injectedService: UserService
+  ) {
+    // Option 1: Create service like this in the resolver
+    this.localService = new BaseService<User>(User, userRepository);
+
+    // Option 2: Create service like this in the resolver
+    this.subclassedService = Container.get(UserService);
+
+    // Option 3: Injected service
+    this.injectedService = injectedService;
```
Comment (goldcaddy77) [inline examples/6-base-service/src/user.resolver.ts]: Options 2 and 3 here use a `UserService` derived from `BaseService`

File: examples/6-base-service/src/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  async create(data: DeepPartial<User>, userId: string): Promise<User> {
+    const newUser = await super.create(data, userId);
+
+    // Perform some side effects
+
+    return newUser;
+  }
+}
```
Comment (goldcaddy77) [inline examples/6-base-service/src/user.service.ts]: By adding the service here, we can isolate side effects without having to put business logic in the resolvers or data models.

File: examples/6-base-service/src/user.resolver.ts (hunk 3 above)
Comment (deweyjose) [reply examples/6-base-service/src/user.resolver.ts]: great! this will make things much more flexible/cleaner when we need flexibility around the DAL

**Pull request merged**